Added Dot2Ct & Dot-Bracket format + implementation in drawrnajs

yhoogstrate opened the pull request:
I added the tool Dot2Ct from the RNAStructure package, including a datatype definition for Dot-Bracket formatted files. The Dot-Bracket format is now also implemented in drawrnajs.

bgruening commented on tools/rna_tools/dot2ct/.shed.yml:
```diff
@@ -0,0 +1,12 @@
+long_description: |
+  Converts the first sequence in a Dot-Bracket file into a CT file
+
+  Party of the RNAStructure package.
```
s/Party/Part/

bgruening commented on tools/rna_tools/dot2ct/datatypes_conf.xml:
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0"?>
+<datatypes>
```
@yhoogstrate can we stick to 4spaces intent for all files? This is for sure debatable, but it is kind of a standard.

bgruening replied:
For more standards please refer to: http://galaxy-iuc-standards.readthedocs.org/en/latest/

bgruening commented on tools/rna_tools/dot2ct/datatypes_conf.xml:
```diff
@@ -0,0 +1,12 @@
+<datatypes>
+	<datatype_files>
+		<datatype_file name="RNAStructure.py"/>
+	</datatype_files>
```
Awesome, we will have today a datatype discussion: https://github.com/galaxy-iuc/standards/issues/13
So we will know soon if we can move this straight into Galaxy main.

bgruening commented on tools/rna_tools/dot2ct/lib/galaxy/datatypes/RNAStructure.py:
```diff
@@ -0,0 +1,93 @@
+from galaxy.datatypes.metadata import MetadataElement
+
+from galaxy.datatypes import metadata
+from galaxy.datatypes import data
```
I think most of the import are not needed, isn't it?

yhoogstrate commented:
Thanks for the feedback. I have addressed the comments above, and I managed to get the visualization button working for the custom DotBracket data type.

bgruening commented:
@yhoogstrate thanks a lot! Will upload it shortly ...
